Java/Java.sublime-syntax

```diff
@@ -8,19 +8,42 @@ file_extensions:
 scope: source.java
 contexts:
   main:
-    - match: '^\s*(package)\b(?:\s*([^ ;$]+)\s*(;)?)?'
-      scope: meta.package.java
-      captures:
-        1: keyword.other.package.java
-        2: storage.modifier.package.java
-        3: punctuation.terminator.java
-    - match: '^\s*(import)\b(?:\s*([^ ;$]+)\s*(;)?)?'
-      scope: meta.import.java
-      captures:
-        1: keyword.other.import.java
-        2: storage.modifier.import.java
-        3: punctuation.terminator.java
+    - include: package
+    - include: import
     - include: code
+  package:
+    - match: '\bpackage\b'
+      scope: keyword.control.package.java
+      push:
+        - meta_scope: meta.package.java
+        - match: '[\w\.]+'
+          scope: keyword.other.package.java
+        - match: ;
+          scope: punctuation.terminator.java
+          pop: true
+        - match: (?=\n)
+          pop: true
+  import:
+    - match: \bimport\b
+      scope: keyword.control.import.java
+      push:
+        - meta_scope: meta.import.java
+        - match: \bstatic\b
+          scope: storage.modifier.static.java
+          push:
+            - match: '[\w\.]+'
+              scope: keyword.other.static.import.java
```

Comment:
I'm open for suggestions. I didn't like the original storage.modifier.package.java because it's not a modifier. It's also not a variable either. It really is a class path. Maybe it should be `storage.type.import.java` ?